File: nextjs-frontend/test_smtp.js
const nodemailer = require('nodemailer');

/**
 * SMTP Test Script for Hostinger
 * Tests the provided SMTP credentials directly
 */

async function testHostingerSMTP() {
  console.log('🧪 SMTP Test Script for Hostinger');
  console.log('=' .repeat(50));
  
  const smtpServer = 'smtp.hostinger.com';
  const senderEmail = 'accounts@intelliglobalconferences.com';
  const password = 'Muni@12345m';
  const receiverEmail = 'professor2004h@gmail.com';
  
  console.log('🔍 Testing Hostinger SMTP Configuration...');
  console.log(`Server: ${smtpServer}`);
  console.log(`Sender: ${senderEmail}`);
  console.log(`Receiver: ${receiverEmail}`);
  
  // Test different port configurations
  const configurations = [
    { port: 465, secure: true, name: 'SSL Port 465' },
    { port: 587, secure: false, name: 'STARTTLS Port 587' },
    { port: 25, secure: false, name: 'Plain Port 25' }
  ];
  
  for (const config of configurations) {
    console.log(`\n📧 Testing ${config.name}...`);
    
    try {
      // Create transporter
      const transporter = nodemailer.createTransport({
        host: smtpServer,
        port: config.port,
        secure: config.secure,
        auth: {
          user: senderEmail,
          pass: password,
        },
        tls: {
          rejectUnauthorized: false
        },
        connectionTimeout: 30000,
        greetingTimeout: 15000,
        socketTimeout: 30000,
        debug: true,
        logger: true
      });
      
      console.log(`  ✓ Transporter created for ${config.name}`);
      
      // Test connection
      console.log('  🔍 Testing connection...');
      await transporter.verify();
      console.log(`  ✅ Connection verified for ${config.name}`);
      
      // Send test email
      console.log('  📧 Sending test email...');
      const mailOptions = {
        from: `"Intelli Global Conferences" <${senderEmail}>`,
        to: receiverEmail,
        subject: `SMTP Test - ${config.name}`,
        html: `
          <h2>SMTP Test Email</h2>
          <p>This email was sent to test the Hostinger SMTP configuration.</p>
          <p><strong>Configuration Details:</strong></p>
          <ul>
            <li>Server: ${smtpServer}</li>
            <li>Port: ${config.port}</li>
            <li>Security: ${config.secure ? 'SSL' : 'STARTTLS'}</li>
            <li>From: ${senderEmail}</li>
            <li>To: ${receiverEmail}</li>
            <li>Timestamp: ${new Date().toISOString()}</li>
          </ul>
          <p>✅ If you receive this email, the SMTP configuration is working!</p>
        `,
        text: `SMTP Test Email - ${config.name}. Timestamp: ${new Date().toISOString()}`
      };
      
      const result = await transporter.sendMail(mailOptions);
      console.log(`  ✅ Email sent successfully via ${config.name}`);
      console.log(`  📧 Message ID: ${result.messageId}`);
      console.log(`  📧 Response: ${result.response}`);
      
      return { success: true, config: config.name, messageId: result.messageId };
      
    } catch (error) {
      console.log(`  ❌ ${config.name} failed:`);
      console.log(`     Error: ${error.message}`);
      console.log(`     Code: ${error.code}`);
      console.log(`     Command: ${error.command}`);
      console.log(`     Response: ${error.response}`);
    }
  }
  
  console.log('\n❌ All SMTP configurations failed');
  return { success: false };
}

async function testBasicConnection() {
  console.log('\n🔍 Testing basic connection to Hostinger SMTP...');
  
  const smtpServer = 'smtp.hostinger.com';
  const ports = [25, 587, 465];
  
  for (const port of ports) {
    try {
      console.log(`  Testing port ${port}...`);
      
      const transporter = nodemailer.createTransport({
        host: smtpServer,
        port: port,
        secure: port === 465,
        connectionTimeout: 10000,
        greetingTimeout: 5000,
        socketTimeout: 10000
      });
      
      await transporter.verify();
      console.log(`  ✓ Successfully connected to ${smtpServer}:${port}`);
      
    } catch (error) {
      console.log(`  ❌ Port ${port} failed: ${error.message}`);
    }
  }
}

async function main() {
  try {
    // Test basic connection first
    await testBasicConnection();
    
    // Test full SMTP with authentication
    const result = await testHostingerSMTP();
    
    if (result.success) {
      console.log('\n🎉 SMTP test completed successfully!');
      console.log(`✅ Working configuration: ${result.config}`);
      console.log(`📧 Message ID: ${result.messageId}`);
    } else {
      console.log('\n💡 Troubleshooting Suggestions:');
      console.log('1. Verify the email account exists in Hostinger control panel');
      console.log('2. Check if SMTP is enabled for the account');
      console.log('3. Confirm the password is correct');
      console.log('4. Check if 2FA is enabled (may need app password)');
      console.log('5. Try creating the email account if it doesn\'t exist');
      console.log('6. Contact Hostinger support for SMTP configuration');
      console.log('7. Check Hostinger documentation for correct SMTP settings');
    }
    
  } catch (error) {
    console.error('❌ Test script failed:', error);
  }
}

// Run the test
main();

File: nextjs-frontend/test_smtp.py
#!/usr/bin/env python3
"""
SMTP Test Script for Hostinger
Tests the provided SMTP credentials directly
"""

import smtplib
import ssl
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

def test_hostinger_smtp():
    # Email configuration
    smtp_server = 'smtp.hostinger.com'
    sender_email = 'accounts@intelliglobalconferences.com'
    password = 'Muni@12345m'
    receiver_email = 'professor2004h@gmail.com'
    
    print("🔍 Testing Hostinger SMTP Configuration...")
    print(f"Server: {smtp_server}")
    print(f"Sender: {sender_email}")
    print(f"Receiver: {receiver_email}")
    
    # Test different port configurations
    configurations = [
        {'port': 465, 'use_ssl': True, 'name': 'SSL Port 465'},
        {'port': 587, 'use_ssl': False, 'name': 'STARTTLS Port 587'},
        {'port': 25, 'use_ssl': False, 'name': 'Plain Port 25'}
    ]
    
    for config in configurations:
        print(f"\n📧 Testing {config['name']}...")
        
        try:
            # Create message
            message = MIMEMultipart()
            message["From"] = sender_email
            message["To"] = receiver_email
            message["Subject"] = f"SMTP Test - {config['name']}"
            
            body = f"""
            SMTP Test Email
            
            Configuration: {config['name']}
            Server: {smtp_server}
            Port: {config['port']}
            SSL: {config['use_ssl']}
            Timestamp: {__import__('datetime').datetime.now()}
            
            If you receive this email, the SMTP configuration is working!
            """
            
            message.attach(MIMEText(body, "plain"))
            
            # Connect and send
            if config['use_ssl']:
                # SSL connection
                context = ssl.create_default_context()
                with smtplib.SMTP_SSL(smtp_server, config['port'], context=context) as server:
                    print(f"  ✓ Connected to {smtp_server}:{config['port']} (SSL)")
                    server.login(sender_email, password)
                    print(f"  ✓ Authenticated as {sender_email}")
                    server.sendmail(sender_email, receiver_email, message.as_string())
                    print(f"  ✅ Email sent successfully via {config['name']}")
                    return True
            else:
                # STARTTLS connection
                with smtplib.SMTP(smtp_server, config['port']) as server:
                    print(f"  ✓ Connected to {smtp_server}:{config['port']}")
                    server.starttls()
                    print(f"  ✓ STARTTLS enabled")
                    server.login(sender_email, password)
                    print(f"  ✓ Authenticated as {sender_email}")
                    server.sendmail(sender_email, receiver_email, message.as_string())
                    print(f"  ✅ Email sent successfully via {config['name']}")
                    return True
                    
        except smtplib.SMTPAuthenticationError as e:
            print(f"  ❌ Authentication failed: {e}")
        except smtplib.SMTPConnectError as e:
            print(f"  ❌ Connection failed: {e}")
        except smtplib.SMTPException as e:
            print(f"  ❌ SMTP error: {e}")
        except Exception as e:
            print(f"  ❌ Unexpected error: {e}")
    
    print("\n❌ All SMTP configurations failed")
    return False

def test_connection_only():
    """Test just the connection without authentication"""
    smtp_server = 'smtp.hostinger.com'
    
    print("\n🔍 Testing basic connection to Hostinger SMTP...")
    
    ports = [25, 587, 465]
    for port in ports:
        try:
            print(f"  Testing port {port}...")
            with smtplib.SMTP(smtp_server, port, timeout=10) as server:
                print(f"  ✓ Successfully connected to {smtp_server}:{port}")
                response = server.noop()
                print(f"  ✓ Server response: {response}")
        except Exception as e:
            print(f"  ❌ Port {port} failed: {e}")

if __name__ == "__main__":
    print("🧪 SMTP Test Script for Hostinger")
    print("=" * 50)
    
    # Test basic connection first
    test_connection_only()
    
    # Test full SMTP with authentication
    success = test_hostinger_smtp()
    
    if success:
        print("\n🎉 SMTP test completed successfully!")
    else:
        print("\n💡 Suggestions:")
        print("1. Verify the email account exists in Hostinger control panel")
        print("2. Check if SMTP is enabled for the account")
        print("3. Confirm the password is correct")
        print("4. Check if 2FA is enabled (may need app password)")
        print("5. Contact Hostinger support for SMTP configuration")
